samples/snippets/csharp/how-to/strings/SearchStrings.cs

```diff
@@ -0,0 +1,158 @@
+
+            // Write the string and include the quotation marks.
+            Console.WriteLine($"\"{factMessage}\"");
+
+            // This search returns the substring between two strings, so 
+            // the first index is moved to the character just after the first string.
+            int first = factMessage.IndexOf("methods") + "methods".Length;
+            int last = factMessage.LastIndexOf("methods");
+            string str2 = factMessage.Substring(first, last - first);
+            Console.WriteLine($"Substring between \"methods\" and \"methods\": '{str2}'");
+
+            /*
+                Output:
+                Substring between "methods" and "methods": ' have all the capabilities of regular static '
+            */
+            // </Snippet2>
+        }
+
+        private static void RegularExpressionsOne()
+        {
+            // <Snippet3>
+            string[] sentences =
+            {
+                "C# code",
+                "Chapter 2: Writing Code",
+                "Unicode",
+                "no match here"
+            };
+
+            string sPattern = "code";
+
+            foreach (string s in sentences)
+            {
+                Console.Write($"{s,24}");
+
+                if (System.Text.RegularExpressions.Regex.IsMatch(s, sPattern, System.Text.RegularExpressions.RegexOptions.IgnoreCase))
+                {
+                    Console.WriteLine($"  (match for '{sPattern}' found)");
+                }
+                else
+                {
+                    Console.WriteLine();
+                }
+            }
+
+            /* Output:
+               C# code  (match for 'code' found)
+               Chapter 2: Writing Code  (match for 'code' found)
+               Unicode  (match for 'code' found)
+               no match here
+            */
+            // </Snippet3>
+        }
+
+        private static void RegularExpressionsValidation()
+        {
+            // <Snippet4>
+            string[] numbers =
+            {
+                "123-555-0190",
+                "444-234-22450",
+                "690-555-0178",
+                "146-893-232",
+                "146-555-0122",
+                "4007-555-0111",
+                "407-555-0111",
+                "407-2-5555",
+                "407-555-8974",
+                "407-2ab-5555",
+                "690-555-8148",
+                "146-893-232-"
+            };
+
+            string sPattern = "^\\d{3}-\\d{3}-\\d{4}$";
+
+            foreach (string s in numbers)
+            {
+                Console.Write($"{s,14}");
+
+                if (System.Text.RegularExpressions.Regex.IsMatch(s, sPattern))
+                {
+                    Console.WriteLine(" - valid");
+                }
+                else
+                {
+                    Console.WriteLine(" - invalid");
+                }
+            }
+
+            /* Output:
```

Comment:
Given the fact that code blocks are interactive and one can click **Run** to see output, should we remove comments with expected outputs from snipppets? There are no those in other two "How to" on strings and I thinks that's fine. If such comments can be removed, then all four methods become much shorter and no need to update comments when code is updated occasionally.